spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/ConformanceCmdConfig.scala

```diff
@@ -0,0 +1,85 @@
+ *
+ * Note: scopt requires all fields to have default values.
+ *       Even if a field is mandatory it needs a default value.
+ */
+case class ConformanceCmdConfig(datasetName: String = "",
+                                datasetVersion: Int = 1,
+                                reportDate: String = "",
+                                reportVersion: Option[Int] = None,
+                                menasCredentialsFactory: MenasCredentialsFactory = InvalidMenasCredentialsFactory,
+                                performanceMetricsFile: Option[String] = None,
+                                folderPrefix: Option[String] = None,
+                                persistStorageLevel: Option[StorageLevel] = None,
+                                publishPathOverride: Option[String] = None,
+                                experimentalMappingRule: Option[Boolean] = None,
+                                isCatalystWorkaroundEnabled: Option[Boolean] = None,
+                                autocleanStandardizedFolder: Option[Boolean] = None)
+  extends ConformanceConfig[ConformanceCmdConfig]
+  with JobCmdConfig[ConformanceCmdConfig]{
+  override def withPublishPathOverride(value: Option[String]): ConformanceCmdConfig = copy(publishPathOverride = value)
+  override def withExperimentalMappingRule(value: Option[Boolean]): ConformanceCmdConfig = copy(experimentalMappingRule = value)
+  override def withIsCatalystWorkaroundEnabled(value: Option[Boolean]): ConformanceCmdConfig = copy(isCatalystWorkaroundEnabled = value)
+  override def withAutocleanStandardizedFolder(value: Option[Boolean]): ConformanceCmdConfig = copy(autocleanStandardizedFolder = value)
+  override def withDatasetName(value: String): ConformanceCmdConfig = copy(datasetName = value)
+  override def withDatasetVersion(value: Int): ConformanceCmdConfig = copy(datasetVersion = value)
+  override def withReportDate(value: String): ConformanceCmdConfig = copy(reportDate = value)
+  override def withReportVersion(value: Option[Int]): ConformanceCmdConfig = copy(reportVersion = value)
+  override def withMenasCredentialsFactory(value: MenasCredentialsFactory): ConformanceCmdConfig = copy(menasCredentialsFactory = value)
+  override def withPerformanceMetricsFile(value: Option[String]): ConformanceCmdConfig = copy(performanceMetricsFile = value)
+  override def withFolderPrefix(value: Option[String]): ConformanceCmdConfig = copy(folderPrefix = value)
+  override def withPersistStorageLevel(value: Option[StorageLevel]): ConformanceCmdConfig = copy(persistStorageLevel = value)
+}
+
+object ConformanceCmdConfig {
+  val stepName = "Conformance"
+
+  def getCmdLineArguments(args: Array[String]): ConformanceCmdConfig = {
+    //Mutually exclusive parameters don't work in scopt 4
+    if (args.contains("--menas-credentials-file") && args.contains("--menas-auth-keytab")) {
+      println("ERROR: Only one authentication method is allowed at a time")
+      System.exit(1)
```

Comment:
Throw some exception or return a `Try[ConformanceCmdConfig]` (that was a CQC consensus for this in Herme)

Reply:
I would do this on #1392

Reply:
ok

Reply:
@benedeki  sorry, is this still valid?